source/module_base/timer.cpp

```diff
@@ -255,6 +259,7 @@ void timer::print_all(std::ofstream &ofs)
 	std::vector<int> calls;
 	std::vector<double> avgs;
 	std::vector<double> pers;
+	const double TIME_EPSILON = 1e-9;
```

Comment:
suggst using time_epsion instead of TIME_EPSILON

Reply:
also, you could add some comments to explain why you choose a 1e-9 (magic number)